general-concepts/ebuild-revisions/text.xml

```diff
@@ -8,37 +8,98 @@
-time; developers should use their best judgement in these circumstances.
+The developers are encouraged to use common sense when determining
+whether to introduce a new <c>-rX</c> revision. The following rule
+of thumb could be used as a guideline:
 </p>
+<ol>
+  <li>
+    If the change is likely to break the package and require users to
+    downgrade (especially if the package is stable, all non-trivial
+    changes are considered as such), then a new revision should be
+    introduced and the old one kept. If the package has stable keywords,
+    the new revision should be dropped to <c>~arch</c>
+    (see <uri link="::keywording#Keywording on Upgrades"/>).
+  </li>
 
-<important>
-For ebuilds marked stable on at least one arch, only trivial edits can be made
-without a bump (e.g. typo fixes in elog messages). Even simple changes may
-result in a breakage. <b>Modifying stable ebuilds should be avoided.</b>
-</important>
+  <li>
+    If the change makes a substantial difference to the user who already
+    installed the package (fixes runtime issues, changes installed files,
+    etc.) and it would not be propagated using other means, then
+    the ebuild should be renamed to a new revision. If the package has
+    stable keywords, they should be kept intact.
+  </li>
 
-<p>
-When doing a revision bump, the usual rules about dropping to <c>~arch</c> apply.
-See <uri link="::keywording#Keywording on Upgrades"/>.
-</p>
+  <li>
+    Otherwise, the change can be done in place in the current revision
+    of the ebuild.
+  </li>
+</ol>
+
+<p>The examples of changes that warrant a new revision are:</p>
+<ul>
+  <li>adding a patch to fix a runtime issue,</li>
+  <li>installing additional files that could be useful to the user,</li>
+  <li>adding a missing runtime dependency to one of the existing blocks,</li>
+  <li>
+    adding a missing build-time dependency that caused the package
+    to be built successfully but incorrectly (e.g. miss some feature),
+  </li>
+  <li>
+    restricting a runtime dependency version, unless the <c>:=</c>
+    subslot operator is going to trigger a rebuild.
+  </li>
+</ul>
+
+<p>The examples of changes that can be done without a revison bump are:</p>
+<ul>
+  <li>
+    adding a patch to fix a build-time issue that prevented users from
+    building the package (since it does not affect users who already
+    managed to build it),
+  </li>
+  <li>adding a trivial documentation fix,</li>
+  <li>
+    installing an addition file of relatively little value (minor
+    documentation, editor syntax file, bash completion) compared
+    to the cost of rebuilding the package (especially if a new bump
+    is expected soon),
+  </li>
+  <li>
+    adding a missing build-time dependency that caused the package to
+    fail to build,
+  </li>
+  <li>
+    adding a new USE flag or removing an existing one (since change
+    in USE flags is going to trigger <c>--changed-use</c> rebuild),
```

Comment:
It also requires that the `--changed-use` behavior be _available_ in the user's package manager, and it isn't in the PMS. This probably isn't the place to get into this, but I would rather see us do revbumps for USE flag changes than try to work around it on the user's end. There are two viable scenarios:

**Scenario 1**: Developers make new revisions whenever they change USE flags. All users immediately see the changes, and no special package manager support is needed.

**Scenario 2**: We document that it's OK for developers to change USE flags without a new revision. We insist that all package managers implement flags like `--newuse` and `--changed-use`. Moreover, we insist that users enable those flags when performing upgrades.

The outcomes of both scenarios are identical, but the second one is a hell of a lot more complicated. We would be better off requiring new revisions for USE changes, and then getting rid of the "required" `--newuse` and `--changed-use` that are littered everywhere.